### PR TITLE
add graalvm example - fixes #152

### LINKED ISSUE #152: Provide an example for compiling to native-image
I fail with and seemingly endless chain of problems when I try to build a native-image with graalvm.
Has anyone setup a working example?

### PR DESCRIPTION
**Pull request by jamesward**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -46,6 +48,19 @@ In another console, run the client against the "dog", "pig", and "sheep" service
...
+
+Start the server:
+```sh
+./gradlew :server:HelloWorldServer
+```
+
+In another console, create the native image client and run it:
+```sh
+./gradlew :native-client:nativeImage
+native-client/build/graal/hello-world
```
Comment (bshaffer) [inline examples/README.md:61]: I received the following error at this step (sorry if I did something stupid):

```
> Task :native-client:nativeImage FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':native-client:nativeImage'.
> A problem occurred starting process 'command '/Users/betterbrent/.gradle/caches/com.palantir.graal/20.2.0/8/graalvm-ce-java8-20.2.0/Contents/Home/bin/native-image''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
14 actionable tasks: 1 executed, 13 up-to-date
```

Comment (jamesward) [reply examples/README.md:61]: Can you rerun with `--info`?

Also, can you verify that you can run `/Users/betterbrent/.gradle/caches/com.palantir.graal/20.2.0/8/graalvm-ce-java8-20.2.0/Contents/Home/bin/native-image` ?

Comment (bshaffer) [reply examples/README.md:61]: "No such file or directory"... well that would indeed throw an error!

Comment (bshaffer) [reply examples/README.md:61]: What is responsible for creating that file?

Comment (jamesward) [reply examples/README.md:61]: The `:native-client:nativeImage` task is supposed to download GraalVM and put it in that location.  Can you check to see what is in `/Users/betterbrent/.gradle/caches/com.palantir.graal` ?

Comment (jamesward) [reply examples/README.md:61]: Note, that it should have downloaded `graalvm-ce-java8-darwin-amd64-20.2.0.tar.gz
` from `https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-20.2.0`

Comment (bshaffer) [reply examples/README.md:61]: The issue is there was something off with the default port (50051) being used (still not sure what it was), but setting `PORT` to a different number worked.

**Review by bshaffer — APPROVED** [review]

**Comment by vyshane** [comment]
I see that the example only demonstrates a client use case. Is GraalVM supported for gRPC servers as well?

**Comment by jamesward** [comment]
I think a server is possible and will try to do that. Can you file a new issue?

**Comment by vyshane** [comment]
Sure, have created an issue :)

**Pull request merged**